sklearnex/linear_model/tests/test_incremental_linear.py

```diff
@@ -0,0 +1,202 @@
+@pytest.mark.parametrize("macro_block", [None, 1024])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_sklearnex_partial_fit_on_gold_data(
+    dataframe, queue, fit_intercept, macro_block, dtype
+):
+    X = np.array([[1], [2], [3], [4]])
+    X = X.astype(dtype=dtype)
+    y = X + 3
+    y = y.astype(dtype=dtype)
+    X_split = np.array_split(X, 2)
+    y_split = np.array_split(y, 2)
+
+    inclin = IncrementalLinearRegression()
+    if macro_block is not None:
+        hparams = inclin.get_hyperparameters("fit")
+        hparams.cpu_macro_block = macro_block
+        hparams.gpu_macro_block = macro_block
+    for i in range(2):
+        X_split_df = _convert_to_dataframe(
+            X_split[i], sycl_queue=queue, target_df=dataframe
+        )
+        y_split_df = _convert_to_dataframe(
+            y_split[i], sycl_queue=queue, target_df=dataframe
+        )
+        inclin.partial_fit(X_split_df, y_split_df)
+
+    assert inclin.n_features_in_ == 1
+    assert_allclose(inclin.coef_, [[1]], rtol=1e-6, atol=1e-6)
+    if fit_intercept:
+        assert_allclose(inclin.intercept_, 3, rtol=1e-6, atol=1e-6)
+
+    X_df = _convert_to_dataframe(X, sycl_queue=queue, target_df=dataframe)
+    y_pred = inclin.predict(X_df)
+    # TODO Implement more clear way to transfer dpctl/dpnp arrays to host
+    _, hostdata = _transfer_to_host(queue, y_pred)
+    y_pred = hostdata[0]
+
+    assert_allclose(y_pred, y, rtol=1e-6, atol=1e-6)
+
+
+@pytest.mark.parametrize("dataframe,queue", get_dataframes_and_queues())
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize("macro_block", [None, 1024])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_sklearnex_partial_fit_multitarget_on_gold_data(
+    dataframe, queue, fit_intercept, macro_block, dtype
+):
+    X = np.array([[1, 1], [1, 2], [2, 2], [2, 3]])
+    X = X.astype(dtype=dtype)
+    y = np.dot(X, [1, 2]) + 3
+    y = y.astype(dtype=dtype)
+    X_split = np.array_split(X, 2)
+    y_split = np.array_split(y, 2)
+
+    inclin = IncrementalLinearRegression()
+    if macro_block is not None:
+        hparams = inclin.get_hyperparameters("fit")
+        hparams.cpu_macro_block = macro_block
+        hparams.gpu_macro_block = macro_block
+    for i in range(2):
+        X_split_df = _convert_to_dataframe(
+            X_split[i], sycl_queue=queue, target_df=dataframe
+        )
+        y_split_df = _convert_to_dataframe(
+            y_split[i], sycl_queue=queue, target_df=dataframe
+        )
+        inclin.partial_fit(X_split_df, y_split_df)
+
+    assert inclin.n_features_in_ == 2
+    assert_allclose(inclin.coef_, [1.0, 2.0], rtol=1e-5, atol=1e-5)
+    if fit_intercept:
+        assert_allclose(inclin.intercept_, 3.0, rtol=1e-5, atol=1e-5)
+
+    X_df = _convert_to_dataframe(X, sycl_queue=queue, target_df=dataframe)
+    y_pred = inclin.predict(X_df)
+    # TODO Implement more clear way to transfer dpctl/dpnp arrays to host
```

Comment:
I am puzzled why this isn't working out of the box... Message me about this

Reply:
in this comment I mean that it would be good to have transfer_to_host method which doesn't return any redundant data and just converts dpctl to numpy array